swarm/network/stream/snapshot_retrieval_test.go

```diff
@@ -132,14 +171,151 @@ var retrievalSimServiceMap = map[string]simulation.ServiceFunc{
+			}
+
+			return r, cleanup, nil
+		},
+	},
+	)
+	defer sim.Close()
+
+	log.Info("Initializing test config", "node count", nodeCount)
+
+	conf := &synctestConfig{}
+	//map of discover ID to indexes of chunks expected at that ID
+	conf.idToChunksMap = make(map[enode.ID][]int)
+	//map of overlay address to discover ID
+	conf.addrToIDMap = make(map[string]enode.ID)
+	//array where the generated chunk hashes will be stored
+	conf.hashes = make([]storage.Address, 0)
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 3*time.Minute)
+	defer cancelSimRun()
+
+	filename := fmt.Sprintf("testing/snapshot_%d.json", nodeCount)
+	err := sim.UploadSnapshot(ctx, filename)
+	if err != nil {
+		return err
+	}
+
+	log.Info("Starting simulation")
+
+	result := sim.Run(ctx, func(ctx context.Context, sim *simulation.Simulation) error {
+		nodeIDs := sim.UpNodeIDs()
+		// first iteration: create addresses
+		for _, n := range nodeIDs {
+			//get the kademlia overlay address from this ID
+			a := n.Bytes()
+			//append it to the array of all overlay addresses
+			conf.addrs = append(conf.addrs, a)
+			//the proximity calculation is on overlay addr,
+			//the p2p/simulations check func triggers on enode.ID,
+			//so we need to know which overlay addr maps to which nodeID
+			conf.addrToIDMap[string(a)] = n
+		}
+
+		// now create random chunks
+		chunks := make([]chunk.Chunk, chunkCount)
 
-The snapshot should have 'streamer' in its service list.
-*/
+		for i := 0; i < chunkCount; i++ {
+			chunks[i] = storage.GenerateRandomChunk(int64(chunkSize))
+			conf.hashes = append(conf.hashes, chunks[i].Address())
+		}
+		log.Debug("random chunks generated, mapping keys to nodes")
+
+		// map addresses to nodes
+		mapKeysToNodes(conf)
+
+		// second iteration: store chunks at the nodes they would be
+		// expected to be
+		log.Debug("storing every chunk at correspondent node store")
+		for _, id := range nodeIDs {
+			// these are the chunks for this node
+			localChunks := conf.idToChunksMap[id]
+			// for every such chunk (which are only indexes)...
+			for _, ch := range localChunks {
+				item, ok := sim.NodeItem(id, bucketKeyStore)
+				if !ok {
+					return fmt.Errorf("Error accessing localstore")
+				}
+				lstore := item.(chunk.Store)
+				// ...get the actual chunk
+				for _, chnk := range chunks {
+					if bytes.Equal(chnk.Address(), conf.hashes[ch]) {
+						// ...and store it in the localstore
+						_, err = lstore.Put(ctx, chunk.ModePutUpload, chnk)
+						if err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+
+		// now try to retrieve every chunk from every node
+		log.Debug("starting retrieval")
+		cnt := 0
+
+	REPEAT:
+		for {
+			for _, id := range nodeIDs {
+				item, ok := sim.NodeItem(id, bucketKeyFileStore)
+				if !ok {
+					return fmt.Errorf("No filestore")
+				}
+				fileStore := item.(*storage.FileStore)
+				for _, chunk := range chunks {
+					reader, _ := fileStore.Retrieve(context.TODO(), chunk.Address())
```

Comment:
checking error ?

Reply:
That API does not return an error, but a `bool` indicating if it is encrypted. Thus ignored